src/lib/ast.ml

```diff
@@ -0,0 +1,90 @@
+module DataFile = struct
+  type t = { id : int; path : string }
+
+  let create id path = { id; path }
+  let id d = d.id
+  let path d = d.path
+  let compare a b = Int.compare a.id b.id
+end
+
+module Leaf = struct
+  type t = {
+    id : int;
+    command : Command.t;
+    inputs : DataFile.t list;
+    outputs : DataFile.t list;
+  }
+
+  let create id command inputs outputs = { id; command; inputs; outputs }
+  let command o = o.command
+  let inputs o = o.inputs
+  let outputs o = o.outputs
+  let id o = o.id
+end
+
+module CommandGroup = struct
+  type t = { name : string; children : Leaf.t list }
+
+  let create name children = { name; children }
+  let name g = g.name
+  let children g = g.children
+end
+
+(* Not yet an actual AST, actually an ASL :) *)
+type t = CommandGroup.t list
+
+let order_command_list (metadata : Frontmatter.t)
```

Comment:
I think it would be good to not add type annotations here and instead write an `ast.mli` file with some docs à la `md.ml` and `block.ml`.